File: tests/fixtures/sample_ts/service.ts
import { User } from "./models";
import { UserRepository } from "./repository";

export class UserService {
  private repo = new UserRepository();

  register(name: string, email: string): User {
    const user = new User(name, email);
    this.repo.add(user);
    return user;
  }

  headcount(): number {
    return this.repo.count();
  }
}

/** Build a default service instance. */
export const makeDefaultService = (): UserService => new UserService();

File: tests/fixtures/sample_python/service.py
"""User service layer coordinating models and the repository."""

from functools import lru_cache

from models import User
from repository import UserRepository


class UserService:
    """Coordinates user registration and lookups."""

    def __init__(self) -> None:
        self.repo = UserRepository()

    def register(self, name: str, email: str) -> User:
        user = User(name, email)
        self.repo.add(user)
        return user

    def headcount(self) -> int:
        return self.repo.count()


@lru_cache
def make_default_service() -> UserService:
    """Build a cached default service instance."""
    return UserService()
